File: LEETCODE/167. 两数之和 II - 输入有序数组.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int left = 0;
        int right = numbers.length - 1;
        while (left < right) {
            if(numbers[left] + numbers[right] < target) {
                left++;
            } else if (numbers[left] + numbers[right]> target) {
                right--;
            } else {
                break;
            }
        }

        return new int[]{left+1, right+1};
    }
}

File: LEETCODE/kotlin/167. 两数之和 II - 输入有序数组.kt
class Solution {
    fun twoSum(numbers: IntArray, target: Int): IntArray {
        var left = 0
        var right = numbers.size - 1

        while (left < right) {
            if (numbers[left] + numbers[right] < target) {
                left++
            } else if (numbers[left] + numbers[right] > target) {
                right--;
            } else {
                break
            }
        }

        return intArrayOf(left + 1, right + 1)
    }
}
